esphome/components/modbus_controller/modbus_controller.h

```
@@ -195,7 +195,7 @@ inline bool coil_from_vector(int coil, const std::vector<uint8_t> &data) {
  */
 template<typename N> N mask_and_shift_by_rightbit(N data, uint32_t mask) {
   auto result = (mask & data);
-  if (result == 0) {
+  if (result == 0 || mask == 0xFFFFFFFF) {
```

Short circuit for default mask 0xFFFFFFFF.